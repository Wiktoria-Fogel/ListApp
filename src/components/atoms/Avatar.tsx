import React from 'react';
import styled from 'styled-components/native';

type AvatarProps = {
  source: string;
};

export const Avatar: React.FC<AvatarProps> = ({source}) => {
  return (
    <Wrapper>
      <AvatarImage source={{uri: source}} />
    </Wrapper>
  );
};

const Wrapper = styled.View`
  height: 60px;
  width: 60px;
  margin-top: 8px;
`;

const AvatarImage = styled.Image`
  height: 100%;
  width: 100%;
  border-radius: 30px;
`;
