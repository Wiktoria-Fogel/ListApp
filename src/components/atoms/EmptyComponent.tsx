import React from 'react';
import styled from 'styled-components/native';
import {COLORS, VerticalSpacer} from '../theme';

type EmptyComponentProps = {
  title: string;
};

export const EmptyComponent: React.FC<EmptyComponentProps> = ({title}) => {
  return (
    <Wrapper>
      <VerticalSpacer height={12} />
      <Title>{title}</Title>
    </Wrapper>
  );
};

const Wrapper = styled.View`
  height: 100%;
  width: 100%;
  align-items: center;
  background-color: antiquewhite;
`;

const Title = styled.Text`
  font-size: 18px;
  font-family: 'Raleway-Bold';
  color: ${COLORS.white};
`;
