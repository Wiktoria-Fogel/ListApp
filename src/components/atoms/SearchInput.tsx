import {TextInputProps, TouchableOpacity} from 'react-native';
import React, {useState} from 'react';
import styled from 'styled-components/native';
import {CancelIcon, MagnifierIcon} from '../../assets/svg/Icons';
import {COLORS} from '../theme';

type SearchInputProps = {
  onClear: () => void;
};

export const SearchInput: React.FC<SearchInputProps & TextInputProps> = ({
  onClear,
  ...other
}) => {
  const [isFocused, setIsFocused] = useState(false);

  return (
    <Wrap isFocused={isFocused || !!other.value}>
      <MagnifierIcon fill={isFocused ? COLORS.borderFocus : COLORS.border} />
      <StyledTextInput
        returnKeyType="done"
        textAlignVertical="center"
        placeholderTextColor={COLORS.grey600}
        {...other}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
      />
      {other.value ? (
        <TouchableOpacity onPress={onClear}>
          <CancelIcon />
        </TouchableOpacity>
      ) : null}
    </Wrap>
  );
};

const Wrap = styled.View<{
  isFocused: boolean;
}>`
  flex-direction: row;
  align-items: center;
  background-color: #ffffff;
  border: 1px solid
    ${({isFocused}) => (isFocused ? COLORS.borderFocus : COLORS.border)};
  border-radius: 8px;
  width: 100%;

  height: 44px;
  padding-left: 12px;
  padding-right: 12px;
  align-items: center;
  justify-content: center;
  height: 50px;
`;

const StyledTextInput = styled.TextInput`
  font-weight: 400;
  color: ${COLORS.grey900};
  padding: 0;
  margin-left: 10px;
  flex: 1;
`;
