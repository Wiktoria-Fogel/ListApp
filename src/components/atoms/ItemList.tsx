import React, {memo} from 'react';
import styled from 'styled-components/native';
import {PublicEvent} from '../../api/models/PublicEvent';
import {COLORS, VerticalSpacer} from '../theme';
import {Avatar} from './Avatar';

interface ItemListProps {
  item: PublicEvent;
}

export const ItemList: React.FC<ItemListProps> = memo(({item}) => {
  return (
    <ItemWrapper key={item.id}>
      <Row>
        <Avatar source={item.actor.avatar_url} />
        <InformationWrapper>
          <Title>{item.actor.display_login}</Title>
          <VerticalSpacer height={4} />
          <Description>{item.repo.name}</Description>
        </InformationWrapper>
      </Row>
    </ItemWrapper>
  );
});

const Row = styled.View`
  flex-direction: row;
  height: 100%;
`;

const ItemWrapper = styled.View`
  padding: 4px 12px;
  height: 86px;
  flex: auto;
  width: 100%;
  border-radius: 12px;
  background-color: ${COLORS.grey800};
`;

const Title = styled.Text`
  font-size: 14px;
  font-family: 'Raleway-Bold';
  color: ${COLORS.white};
`;

const Description = styled.Text`
  font-size: 12px;
  font-family: 'Raleway-Medium';
  color: ${COLORS.white};
`;

const InformationWrapper = styled.View`
  padding: 8px 12px;
  flex: 1;
`;
