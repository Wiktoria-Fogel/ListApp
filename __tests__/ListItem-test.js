import React from 'react';
import renderer, {act} from 'react-test-renderer';
import {ListItem} from '../src/components/atoms/ListItem';

const item = {
  id: '',
  type: 'test',
  actor: {
    id: 123,
    login: 'test',
    display_login: 'test',
    gravatar_id: '',
    url: '',
    avatar_url: '',
  },
  repo: {
    id: 123,
    name: 'test',
    url: '',
  },
};

test('SearchInput renders correctly', async () => {
  const result = renderer.create(<ListItem item={item} />).toJSON();
  await act(async () => {
    expect(result).toMatchSnapshot();
    return;
  });
});
