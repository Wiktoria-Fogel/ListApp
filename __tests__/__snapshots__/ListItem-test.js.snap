// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#242629",
        "borderRadius": 12,
        "flexBasis": "auto",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 86,
        "paddingBottom": 4,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 4,
        "width": "100%",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 60,
            "marginTop": 8,
            "width": 60,
          },
        ]
      }
    >
      <Image
        source={
          Object {
            "uri": "",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 30,
              "height": "100%",
              "width": "100%",
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Raleway-Bold",
              "fontSize": 14,
            },
          ]
        }
      >
        test
      </Text>
      <View
        height={4}
        style={
          Array [
            Object {
              "height": 4,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Raleway-Medium",
              "fontSize": 12,
            },
          ]
        }
      >
        test
      </Text>
    </View>
  </View>
</View>
`;
